api/v1alpha3/conditions_consts.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+
+const (
+	MachineConditionCount      = 2
+	ControlPlaneConditionCount = 3
+	ClusterConditionCount      = 2
+)
+
+// AzureCluster Conditions and Reasons
+const (
+	NetworkInfrastructureReadyCondition = "NetworkInfrastructureReady"
```

Comment:
FYI in CAPI project we are commenting all the public const used for conditions.
e.g. https://github.com/kubernetes-sigs/cluster-api/blob/53e98d22ccc1510a18b2caead34f4a8014a29983/api/v1alpha3/condition_consts.go#L57-L73

Reply:
FYI in CAPA we decided to provide user a better visibility on what's going on during the Network infrastructure provisioning because this operation might take some time and we want to give Users a perception that things are progressing. This is still WIP, but I do expect that we will document ongoing steps (VPCProvisioning, SecurityGroups provisioning, Elastic IP provisioning etc.)

I don't know if a similar consideration apply to CAPZ as well (network infrastructure provisioning takes time)

Reply:
I think we should make these more granular as well (eg. PublicLoadBalancerReadyCondition, VirtualNetworkReadyCondition, etc.). I don't mind us doing this in a later PR though if you think it's easier to start with this for now @nader-ziada.

Reply:
I'll see if I can do that now

Reply:
not sure how I can make these more granular since the code doesn't check for each step?